docs/imageMgmtAPI.rst

```diff
@@ -0,0 +1,337 @@
+| ``activePlan``  | ``boolean`` | ``True`` to mark the plan active, ``False`` otherwise.  |
++-----------------+-------------+---------------------------------------------------------+
+| ``imageRampups``|  ``List``   | List of rampup definitions as shown below               |
++-----------------+-------------+---------------------------------------------------------+
+
+**Rampup Definition json block:**
+
+.. _rampup-definition:
+
+.. code-block:: json
+
+     {
+      "imageVersion": "string",
+      "rampupPercentage": "int",
+      "stability_tag": "enum"   // [experimental/stable/unstable]
+      }
+
+Refer to `usage of this API <#use-case-image-rampup>`_ for implementing a Canary.
+
+.. _get-rampup-plan:
+
+Get an existing Image Rampup Plan
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Returns a list of matching rampup plans for a given imageType
+
+- **Method:** GET
+- **Request URL:** /imageRampup
+- **Request Body:**
+
++-----------------+-------------+---------------------------------------------------------+
+|   Field Name    |     Type    |            Description                                  |
++=================+=============+=========================================================+
+| ``imageType``   | ``String``  | Image type                                              |
++-----------------+-------------+---------------------------------------------------------+
+
+**Example:**
+
+.. code-block::
+
+     GET /imageRampup?imageType=spark_job
+
+.. _update-rampup-plan:
+
+Update an existing Image Rampup Plan
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Update the active rampup plan and rampup details for an image type.
+
+- **Method:** PATCH
+- **Request URL:** /imageRampup/{planName}
+- **Request Body:**
+
++-----------------+-------------+---------------------------------------------------------+
+|   Field Name    |     Type    |            Description                                  |
++=================+=============+=========================================================+
+| ``activePlan``  | ``boolean`` | ``True`` to mark the plan active, ``False`` otherwise.  |
++-----------------+-------------+---------------------------------------------------------+
+| ``imageRampups``|  ``List``   | List of `rampup definitions <#rampup-definition>`_      |
++-----------------+-------------+---------------------------------------------------------+
+
+Use-Cases and Workflows
+-----------------------
+
+New Jobtype/Platform Binary
+***************************
+When a jobtype/platform developer wants to publish their binary for the 1st time, they need to
+register their **imageType** with Azkaban clusters. Hence, the API to `create image type <#create-image-type>`_
+should be called with the required details.
+
+Following this, a `version of the image type must be published <#create-image-version>`_ to be used within Azkaban.
+This is an Admin API and the privileges for this API are with the Azkaban team.
```

Comment:
Apart from Azkaban ADMIN, Image type owner with role as ADMIN or MEMBER should be able to publish a new version of an image type.